src/litserve/server.py

```diff
@@ -583,6 +589,12 @@ def setup_server(self):
         async def index(request: Request) -> Response:
             return Response(content="litserve running")
 
+        @self.app.get("/health", dependencies=[Depends(self.setup_auth())])
+        async def indexh(request: Request) -> Response:
+            if all(self.workers_setup_status.values()):
```

Comment:
we should cache `all(self.workers_setup_status.values())` once it turn `True` to avoid multiprocessing synchronization.

Reply:
Thank you, @aniketmaurya, for the feedback. I will review it and let you know if I have any questions.

Reply:
can use [functools.cache](https://docs.python.org/3/library/functools.html#functools.cache)

Reply:
for test, I think we can add in `test_simple.py` like [this](https://github.com/Lightning-AI/LitServe/blob/16efafe2144253f4350ee75eaf93cadc088d586a/tests/test_simple.py#L43)